cpg_workflows/large_cohort/combiner.py

```diff
@@ -1,132 +1,129 @@
-    elif sequencing_type == 'genome':
-        params.setdefault('use_genome_default_intervals', True)
-    else:
-        raise ValueError(
-            'Either combiner/intervals must be set, or workflow/sequencing_type must be one of: "exome", "genome"',
-        )
+def _get_samples_from_vds(input_vds: str) -> list[str]:
+    return read_vds(input_vds).variant_data.s.collect()
 
-    sequencing_group_names = [s.id for s in sequencing_groups]
-    logging.info(
-        f'Combining {len(sequencing_groups)} sequencing groups: '
-        f'{", ".join(sequencing_group_names)}, using parameters: '
-        f'{params}',
+
+def _parse_metamist_cohort_output(cohort_query_output: dict[str, Any]) -> list[dict[str, Any]]:
+    return [sg for sg in cohort_query_output["cohorts"][0]["sequencingGroups"]]
+
+
+def _parse_metamist_analysis_output(vds_analysis_query_output: dict[str, Any]) -> list[str]:
+    return [analysis["output"] for analysis in vds_analysis_query_output["analyses"]]
+
+
+def run_combiner(
+    output_vds_path: Path,
+    sequencing_type: str,
+    cohort: str,
+    tmp_prfx: str,
+    existing_vds_ids: int | None = None,
+) -> VariantDataset:
+    input_vds_samples: list[str] = []
+    use_genome_default_intervals: bool = False
+    use_exome_default_intervals: bool = False
+
+    existing_vds: list[str] | None = None
+    project_gvcf_paths: list[str]
+
+    cohort_query_results: dict[str, Any] = query(
+        GET_COHORT_QUERY,
+        variables={
+            "cohort": cohort,
+        },
     )
 
-    gvcf_paths = [str(s.gvcf.path) for s in sequencing_groups if s.gvcf]
-    if not gvcf_paths:
-        raise ValueError('No sequencing groups with GVCFs found')
-
-    logging.info(f'Combining {len(sequencing_group_names)} sequencing groups: {", ".join(sequencing_group_names)}')
-
-    check_duplicates(sequencing_group_names)
-    check_duplicates(gvcf_paths)
-
-    combiner = hl.vds.new_combiner(
-        gvcf_paths=gvcf_paths,
-        gvcf_sample_names=sequencing_group_names,
-        # Header must be used with gvcf_sample_names, otherwise gvcf_sample_names
-        # will be ignored. The first gvcf path works fine as a header because it will
-        # be only read until the last line that begins with "#":
-        gvcf_external_header=gvcf_paths[0],
-        output_path=str(out_vds_path),
-        reference_genome='GRCh38',
-        temp_path=str(tmp_prefix),
+    init_batch()
+    # Only support a single input cohort
+    formatted_query_results: list[dict[str, Any]] = _parse_metamist_cohort_output(cohort_query_results)
+
+    if existing_vds_ids:
+        vds_query_results: dict[str, Any] = query(
+            GET_VDS_ANALYSIS_QUERY,
+            variables={
+                "vds_ids": existing_vds_ids,
+            },
+        )
+        existing_vds = _parse_metamist_analysis_output(vds_query_results)
+        input_vds_samples = list(chain(input_vds_samples, *[_get_samples_from_vds(vds) for vds in existing_vds]))
```

Comment:
I guess in theory you can merge as many VDSs as you want, but would this always be an increment based on the latest VDS, or is there a scenario where you'd want a union across many?

Reply:
This is mostly just for the base case where you might have generated a bunch of VDSs already, and want to merge them all. Going forward from that point it'd be almost always the case that you'd be adding some extra samples to an existing VDS.